winit/src/system.rs

```diff
@@ -0,0 +1,53 @@
+//! Access the native system.
+use crate::command::{self, Command};
+pub use iced_native::system::*;
+
+use iced_graphics::window;
+
+/// Query for available system information.
+///
+/// Returns `None` if not using the `sysinfo` feature flag.
+pub fn information<Message>(
+    f: impl Fn(Option<Information>) -> Message + 'static,
+) -> Command<Message> {
+    Command::single(command::Action::System(Action::QueryInformation(
+        Box::new(f),
+    )))
+}
+
+#[cfg(feature = "sysinfo")]
+pub(crate) fn get_information(
+    graphics_info: &window::GraphicsInformation,
+) -> Option<Information> {
+    use sysinfo::{ProcessExt, ProcessorExt, System, SystemExt};
+    let mut system = System::new_all();
+    system.refresh_all();
+
+    let cpu = system.global_processor_info();
+
+    let memory_used = sysinfo::get_current_pid()
+        .and_then(|pid| system.process(pid).ok_or("Process not found"))
+        .and_then(|process| Ok(process.memory()))
+        .ok();
+
+    let information = Information {
+        system_name: system.name(),
+        system_kernel: system.kernel_version(),
+        system_version: system.long_os_version(),
+        cpu_brand: cpu.brand().into(),
+        cpu_cores: system.physical_core_count(),
+        memory_total: system.total_memory(),
+        memory_used,
+        graphics_adapter: graphics_info.adapter.clone(),
+        graphics_backend: graphics_info.backend.clone(),
+    };
+
+    Some(information)
+}
+
+#[cfg(not(feature = "sysinfo"))]
+pub(crate) fn get_information(
+    _graphics_info: &window::GraphicsInformation,
+) -> Option<Information> {
+    None
```

Comment:
I wasn't sure about removing the `system::Information` struct or/and removing the `System::QueryInformation` enum variant if `sysinfo` wasn't enabled, so I decided to turn this into an `Option`.

Any improvements on this would be nice!

Reply:
Simply conditionally compiling the `system::information` helper wouldn't work
because the `run_command` helper matches on every variant of `Action` and so
would throw the error for anyone not using `sysinfo` feature.

So, I've gone ahead and completely remove the `sysinfo` flag in https://github.com/iced-rs/iced/pull/1314/commits/e6b09e9194011c2a588cab0d9f1af1b9a35da787.
By not storing and delaying the fetching of information until requested,
we remove the need of feature flagging.

Reverting this should be as simples as yanking the last commit.

Reply:
Gave it a shot in 93bfe2c75ec97ef78f993926c703f040dde4a5f3.

Reply:
Interesting, would it be possible to also gate the `iced_native` side? That would make it impossible for someone to misuse this.

I've also commited a simple fix to another issue in https://github.com/iced-rs/iced/pull/1314/commits/a4477723d44c35effc8f4bb0ff0cfce93fdf4695 (which I was planning on squashing, but didn't for now).